src/layouts/PLayoutGrid/PLayoutGrid.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="p-layout-grid">
+    <div class="p-layout-grid__header">
+      <slot name="header">
+        <div class="p-layout-grid__header-leading">
+          <slot name="header-leading" />
+        </div>
+        <div class="p-layout-grid__header-trailing">
+          <slot name="header-trailing" />
+        </div>
+      </slot>
+    </div>
+    <div class="p-layout-grid__toolbar">
+      <slot name="toolbar">
+        <div class="p-layout-grid__toolbar-leading">
+          <slot name="toolbar-leading" />
+        </div>
+        <div class="p-layout-grid__toolbar-trailing">
+          <slot name="toolbar-trailing" />
+        </div>
+      </slot>
+    </div>
+    <main class="p-layout-grid__main">
+      <div v-if="$slots.leftRail" class="p-layout-grid__left-rail">
+        <slot name="leftRail" />
+      </div>
+      <div class="p-layout-grid__content">
+        <slot />
+      </div>
+      <div v-if="$slots.rightRail" class="p-layout-grid__right-rail">
+        <slot name="rightRail" />
+      </div>
+    </main>
+  </div>
+</template>
+
+ <style>
+  .p-layout-grid {
+    @apply
+    w-full
+    h-full
+    border-l
+    flex
+    flex-col
+    border-[color:var(--p-color-divider)];
+  }
+
+  .p-layout-grid__toolbar,
+  .p-layout-grid__header {
+    @apply
+    h-14
+    min-h-14
+    border-b
+    border-[color:var(--p-color-divider)]
+    flex
+    justify-between
+    pl-8
+    pr-4
+  }
+
+  .p-layout-grid__toolbar {
+    @apply
+    h-12
+    min-h-12
+  }
+
+  .p-layout-grid__toolbar-leading,
+  .p-layout-grid__header-leading {
+    @apply
+    self-center
+    flex
+    flex-none
+  }
+
+  .p-layout-grid__toolbar-trailing,
+  .p-layout-grid__header-trailing {
+    @apply
+    self-center
+    flex
+    flex-none
+    justify-end
+  }
+
+
+  .p-layout-grid__main {
+    @apply
+    flex
+    flex-1
+    w-full
+    relative
+    flex-nowrap;
+  }
+
+  .p-layout-grid__content {
+    @apply
+    flex-1
+    border-[color:var(--p-color-divider)];
```

Comment:
This _might_ be intentional but I think the bug in the demo exposed that this border still shows up even when you've not provided rail slots:

![image](https://github.com/PrefectHQ/prefect-design/assets/27291717/c9d9274e-3013-4138-baa7-72c59624a978)